File: app/domain/strategies/pagar_en_sitio_strategy.rb
require_relative '../interfaces/i_pago_strategy'

module Strategies
  class PagarEnSitioStrategy
    include Interfaces::IPagoStrategy

    def nombre
      "PagarEnSitio"
    end

    def procesar(monto, cliente_id)
      puts "Pago en efectivo programado para #{monto}. Por favor cumpla no sea ratón."
      true # Se asume que el pago se realizará
    end
  end
end

File: EduLink.Domain/Strategies/PagarEnSitioStrategy.cs
using EduLink.Domain.Interfaces;
using System;
using System.Threading.Tasks;

namespace EduLink.Domain.Strategies;

public class PagarEnSitioStrategy : IPagoStrategy
{
	public string Nombre => "PagarEnSitio";
	public Task<bool> ProcesarAsync(decimal monto, Guid clienteId)
	{
		Console.WriteLine($"Pago en efectivo programado para ${monto:F2}. Por favor cumpla no sea ratón.");
		return Task.FromResult(true); // Siempre debería marcarse como exitosoid
	}
}